test/acceptance/test/profiles_test.go

```diff
@@ -0,0 +1,171 @@
+package acceptance
+
+import (
+	"context"
+	"encoding/base64"
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"net/url"
+	"time"
+
+	sourcev1beta1 "github.com/fluxcd/source-controller/api/v1beta1"
+	pb "github.com/weaveworks/weave-gitops/pkg/api/profiles"
+	"github.com/weaveworks/weave-gitops/pkg/gitproviders"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	"path/filepath"
+
+	"k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/tools/clientcmd"
+	"k8s.io/client-go/util/homedir"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+)
+
+var _ = Describe("Weave GitOps Profiles API", func() {
+	var (
+		namespace        = "test-namespace"
+		appRepoRemoteURL string
+		tip              TestInputs
+		profilesSvc      = "profiles-server"
+		clientSet        *kubernetes.Clientset
+		kClient          client.Client
+	)
+
+	BeforeEach(func() {
+		Expect(FileExists(WEGO_BIN_PATH)).To(BeTrue())
+		Expect(GITHUB_ORG).NotTo(BeEmpty())
+
+		_, _, err := ResetOrCreateCluster(namespace, true)
+		Expect(err).NotTo(HaveOccurred())
+
+		private := true
+		tip = generateTestInputs()
+		_ = initAndCreateEmptyRepo(tip.appRepoName, gitproviders.GitProviderGitHub, private, GITHUB_ORG)
+
+		clientSet, kClient = buildKubernetesClients()
+	})
+
+	AfterEach(func() {
+		deleteRepo(tip.appRepoName, gitproviders.GitProviderGitHub, GITHUB_ORG)
+	})
+
+	It("Works", func() {
```

Comment:
Maybe a little more description on this test case? What are we trying to prove/validate with this test?

Reply:
updated